### ISSUE TITLE
Support additional palette formats

### ISSUE THREAD

**Issue by warrengalyen** [PixiEditor/PixiEditor#586]
I would like to volunteer myself to add the following palette format support:

1. Adobe Photoshop Color Swatches (.aco) - READ/WRITE
2. DeluxePaint (.lbm/.bbm) [yes, the old school bitmap graphics editor] - READ
3. CorelDraw (.pal) - READ

**Comment by flabbet** [comment]
Hey, thanks for volunteering! Sure, feel free to implement those. There might be one issue though, current system isn't prepared for multiple formats containing same extension, we already have implemented `.pal` format, which is JASC Palette format. You'll have to adjust parser resolver to handle this case. If you encounter any troubles, feel free join our Discord https://discord.gg/qSRMYmq for quick response

**Comment by warrengalyen** [comment]
@flabbet Yeah, might have to adjust the import logic a bit for palettes. There are quite a few formats older formats out there use the .pal extension. I would suggest adding a simple function like `CanReadFrom` to the `PaletteFileParser` abstract class that does a quick byte check of the header.

In other news, I have .aco import working with conversion of all supported file color spaces (RGB HSB, LAB, CMYK, Grayscale). I was attempting to test write support but don't see where this is incorporated yet within the program outside the class function. Sorry, this is my first time working with this code base.

**Comment by flabbet** [comment]
`CanReadFrom` is a good idea. About .aco import/export. I feel like adjusting palettes system just for that format, so it supports multiple collections per palette is  a bit too much. I suggest either flattening all colors into one palette or import all collections as a separate palettes into Palette Browser and then apply first one into current palette. I think that first solution is both easier to implement and more consistent. @Equbuxu what do you think?

**Comment by warrengalyen** [comment]
@flabbet Importing all the colors to a single collection would definitely be the easiest since it wouldn't require modifying the import logic. I agree, modifying the palette system is really unnecessary for a single use case.
